### PR TITLE
feat(eks): imported kubectl provider for imported clusters

### PR DESCRIPTION
**Pull request by aka-toxa**
This resolves [issue#12107](https://github.com/aws/aws-cdk/issues/12107) 
we bring the ability to provide existing kubectl provider to the imported cluster
so we can create k8s kubectl role and tight it's trusted entity to single lambda and pass this single lambda to all cdk stacks that works with imported clusters

@iliapolo can you please take a look on this if this approach is fine? if it is I will add documentation and tests 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by aka-toxa** [comment]
so user will be able to call 

```ts
const provider = KubectlProvider.fromKubectlProviderAttributes(...)
const cluster = Cluster.fromClusterAttributes(..., { kubectlprovider: provider })
```

**Comment by iliapolo** [comment]
@aka-toxa Sorry for the delay. I like this approach. You still up to follow it through?

**Review by iliapolo — DISMISSED** [review]
Marking as changes requested. I'm good with the approach - as @aka-toxa mentioned, we need docs and tests.

**Comment by aka-toxa** [comment]
hi guys 
finally, get some time to finish this. 
starting finalising now, hopefully next week we will get something

**Comment by aka-toxa** [comment]
hey @iliapolo it looks like I have all tests and docs now can you review this again please?

**Comment by aka-toxa** [comment]
hi @iliapolo any chances to make a review of this this week?

**Comment by iliapolo** [comment]
> hi @iliapolo any chances to make a review of this week?

I'll try making some time for it 👍

File: packages/@aws-cdk/aws-eks/test/integ.eks-kubectl-provider.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/// !cdk-integ pragma:ignore-assets
```
[the remark this answers is not in the document]
Comment (aka-toxa) [reply packages/@aws-cdk/aws-eks/test/integ.eks-kubectl-provider.ts]: that was obsolete test that I used to debug things actually

File: packages/@aws-cdk/aws-eks/README.md [hunk 2]
```diff
@@ -554,6 +554,22 @@ const cluster = new eks.Cluster(this, 'hello-eks', {
...
+
+```ts
+const handlerRole = iam.Role.fromRoleArn(stack, 'HandlerRole', 'arn:aws:iam::123456789012:role/lambda-role');
+const kubectlProvider = KubectlProvider.fromKubectlProviderAttributes(stack, 'KubectlProvider', {
+  functionArn: 'arn:aws:lambda:us-east-2:123456789012:function:my-function:1',
+  kubectlRoleArn: 'arn:aws:iam::123456789012:role/kubectl-role',
+  handlerRole,
+});
+
+const cluster = eks.Cluster.fromClusterAttributes(stack, 'Cluster', {
```
[the remark this answers is not in the document]
Comment (aka-toxa) [reply packages/@aws-cdk/aws-eks/README.md:588]: done

**Comment by aka-toxa** [comment]
@iliapolo looks like I've addressed your comments, can we merge this?

**Review by iliapolo — DISMISSED** [review]
@aka-toxa What about other usages of `KubectlProvider.getOrCreate`?

I see there are still many places that will create a provider regardless of whether one was provided or not:

- https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-eks/lib/helm-chart.ts#L98
- https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-eks/lib/k8s-patch.ts#L79
- https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-eks/lib/k8s-object-value.ts#L69

I think a better solution would be to change the `getOrCreate` implementation itself, to return the provider on the cluster if one was passed.

**Comment by markussiebert** [comment]
Can I support somehow? Would like to see this feature in an upcoming version :-)

**Comment by iliapolo** [comment]
> Can I support somehow? Would like to see this feature in an upcoming version :-)

@markussiebert is you're willing to pick this up that would be great. Its really close, just need to update some usages of `KubectlProvider.getOrCreate` as mentioned [here](https://github.com/aws/aws-cdk/pull/14689#pullrequestreview-806301557)

**Comment by markussiebert** [comment]
yes, 4 sure, but i don't know how i can contribute to an existing PR? Should i create a new one on the base of my form with the changes included in here? It would be a new PR...

**Comment by iliapolo** [comment]
> yes, 4 sure, but i don't know how i can contribute to an existing PR? Should i create a new one on the base of my fork with the changes from this pr included? It would be a new PR...

Yeap, a new PR is great 👍

**Comment by aka-toxa** [comment]
@markussiebert I can give you permissions

**Comment by markussiebert** [comment]
@aka-toxa thank you, will look into this this evening

**Comment by markussiebert** [comment]
> @aka-toxa What about other usages of `KubectlProvider.getOrCreate`?
> 
> I see there are still many places that will create a provider regardless of whether one was provided or not:
> 
> * https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-eks/lib/helm-chart.ts#L98
> * https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-eks/lib/k8s-patch.ts#L79
> * https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-eks/lib/k8s-object-value.ts#L69
> 
> I think a better solution would be to change the `getOrCreate` implementation itself, to return the provider on the cluster if one was passed.

I think it should now work for every case ... I tried to write a test for this trying to ensure the correct tokens/arns and that there will be no kubectlprovider created.

so feel free to review and let me know what to change :-)

**Comment by markussiebert** [comment]
@aka-toxa can you please ask for a new review from iliapolo

**Comment by aka-toxa** [comment]
@markussiebert asked. thanks for your contribution! I appreciate this

**Comment by markussiebert** [comment]
@iliapolo can you please take a look at this?

**Review by iliapolo — APPROVED** [review]

**Review by iliapolo — DISMISSED** [review]

**Review by iliapolo — APPROVED** [review]

**Pull request merged**